container/container.go

```diff
@@ -667,15 +667,30 @@ func (container *Container) BuildCreateEndpointOptions(n libnetwork.Network, epC
 
 	if epConfig != nil {
 		ipam := epConfig.IPAMConfig
-		if ipam != nil && (ipam.IPv4Address != "" || ipam.IPv6Address != "" || len(ipam.LinkLocalIPs) > 0) {
+
+		if ipam != nil {
 			var ipList []net.IP
+			var ip, ip6, linkip net.IP
```

Comment:
Thanks @chchliang 
I was also suggesting to define all these vars in a 
```
var (

)
```
block